Short-Answer/Algorithms_Answers.md

```diff
@@ -2,14 +2,23 @@
 
 ## Exercise I
 
-a)
+a) setting variable and running basic loop runs at constant time
+Big O of N
+O(n)
 
 
-b)
+b) setting variable and running nested loops (n*n)
+Big O of N Squared
```

Comment:
For this one we have to consider both loops individually:
The outer for loop will run n times, since it is incrementing i by 1 each iteration. This is O(n).
The inner loop is doubling the value of j every time it runs. This means that the number of times the inner while loop runs is not growing at a linear rate. This is O(log(n)).
Since these loops are nested, we multiply the individual runtimes of each loop to get the runtime of the entire block.
So the answer is O(n log(n))